### ISSUE TITLE
harbor v2.1.1 upgrade v2.3.1  harbor-db   cannot access '/var/lib/postgresql/data': Operation not permitted

### ISSUE THREAD

**Issue by xmskf** [goharbor/harbor#15464]
## system info

```shell
Operating system ： 
CentOS Linux release 7.6.1810 (Core)
docker version :
Docker version 19.03.12, build 48a66213fe
docker-compose verion:
docker-compose version 1.29.2, build 5becea4c
```

harbor.yml
```shell
# Harbor DB configuration
database:
  # The password for the root user of Harbor DB. Change this before any production use.
  password: *********
  # The maximum number of connections in the idle connection pool. If it <=0, no idle connections are retained.
  max_idle_conns: 100
  # The maximum number of open connections to the database. If it <= 0, then there is no limit on the number of open connections.
  # Note: the default number of connections is 1024 for postgres of harbor.
  max_open_conns: 900

# The default data volume
data_volume: /data
```

## filesystem

```shell
[root@harbor database]# pwd
/data/database
[root@harbor database]# ll -h
total 56K
drwx------ 19 polkitd input 4.0K Aug 20 11:16 pg13
drwx------ 9 polkitd input  129 Aug  4 15:21 base
drwx------ 2 polkitd input 4.0K Aug 20 16:13 global
drwx------ 2 polkitd input   74 Aug  4 15:21 pg_clog
drwx------ 2 polkitd input   10 Aug  4 15:21 pg_commit_ts
drwx------ 2 polkitd input   10 Aug  4 15:21 pg_dynshmem
-rw------- 1 polkitd input 4.4K Aug  4 15:21 pg_hba.conf
-rw------- 1 polkitd input 1.6K Aug  4 15:21 pg_ident.conf
drwx------ 4 polkitd input   51 Aug  4 15:21 pg_logical
drwx------ 4 polkitd input   48 Aug  4 15:21 pg_multixact
drwx------ 2 polkitd input   26 Aug 20 16:13 pg_notify
drwx------ 2 polkitd input   10 Aug  4 15:21 pg_replslot
drwx------ 2 polkitd input   10 Aug  4 15:21 pg_serial
drwx------ 2 polkitd input   10 Aug  4 15:21 pg_snapshots
drwx------ 2 polkitd input   10 Aug 20 16:13 pg_stat
drwx------ 2 polkitd input  154 Aug 20 16:21 pg_stat_tmp
drwx------ 2 polkitd input   26 Aug 19 15:17 pg_subtrans
drwx------ 2 polkitd input   10 Aug  4 15:21 pg_tblspc
drwx------ 2 polkitd input   10 Aug  4 15:21 pg_twophase
-rw------- 1 polkitd input    4 Aug  4 15:21 PG_VERSION
drwx------ 3 polkitd input  216 Aug 20 09:07 pg_xlog
-rw------- 1 polkitd input   88 Aug  4 15:21 postgresql.auto.conf
-rw------- 1 polkitd input  22K Aug  4 15:21 postgresql.conf
-rw------- 1 polkitd input   78 Aug 20 16:13 postmaster.opts
```

## Problem

When I upgraded from harbor v2.1.1 to v2.3.1, harbor-db Cannot start normally, the log prompts as follows:
![image](https://user-images.githubusercontent.com/6941905/130204715-5399bc9b-cad2-493b-a3dc-71b6fdd4dcbb.png)

When I removed  ”/data/database/pg13“ , the harbor started, but there was no data. There are still exceptions inside the harbor-db container:
![image](https://user-images.githubusercontent.com/6941905/130204107-b93f1061-d402-4cc0-b7c4-1a64804ecfe2.png)
![image](https://user-images.githubusercontent.com/6941905/130204228-e4934ce3-c456-49f7-a92e-116cb753168f.png)

Can someone help me check it out? Thank you!

**Comment by stonezdj** [comment]
It seems that the user is changed from postgres to polkitd, you should run prepare and restart harbor.
```
./prepare --with-notary --with-chartmuseum --trivy
docker-compose down -v 
docker-compose up -d
```

**Comment by xmskf** [comment]
I tried this method, but to no avail.

polkitd:input, this is the file system of the host machine (docker mount point).
 
The way I upgrade harbor is:
https://goharbor.io/docs/2.3.0/administration/upgrade/

**Comment by wy65701436** [comment]
hi, if the prepare cannot help to resolve permission issue on centos, you have to do it manually. 

Please make sure all of files(include the database directory itself) under database directory can assigned to uid 999 & gid 999. Then to remove the pg13 directory and restart harbor.

**Comment by xmskf** [comment]
thanks for your help!   I will try again next time.
I already upgraded to v2.1.6 and will not upgrade v2.3.1 for the time being.

**Comment by kasenhoo** [comment]
> hi, if the prepare cannot help to resolve permission issue on centos, you have to do it manually.
> 
> Please make sure all of files(include the database directory itself) under database directory is assigned to uid 999 & gid 999. Then to remove the pg13 directory and restart harbor.
> 
> Or another workaroud for this: set `privileged: true` for postgresql in the docker compose yml file.

I tried with the command,,,

but failed....revert to 2.1.6

**Comment by wy65701436** [comment]
> > hi, if the prepare cannot help to resolve permission issue on centos, you have to do it manually.
> > Please make sure all of files(include the database directory itself) under database directory is assigned to uid 999 & gid 999. Then to remove the pg13 directory and restart harbor.
> > Or another workaroud for this: set `privileged: true` for postgresql in the docker compose yml file.
> 
> I tried with the command,,,
> 
> but failed....revert to 2.1.6

did you see the permission error in the postgres.log? and did you try to add `privileged: true` in the docker-compose.yml?

**Comment by jorgemansillacriado** [comment]
> > > hi, if the prepare cannot help to resolve permission issue on centos, you have to do it manually.
> > > Please make sure all of files(include the database directory itself) under database directory is assigned to uid 999 & gid 999. Then to remove the pg13 directory and restart harbor.
> > > Or another workaroud for this: set `privileged: true` for postgresql in the docker compose yml file.
> > 
> > 
> > I tried with the command,,,
> > but failed....revert to 2.1.6
> 
> did you see the permission error in the postgres.log? and did you try to add `privileged: true` in the docker-compose.yml?

I have put privileged: true and docker-compose down and up is working ok now

**Comment by dajeffers** [comment]
I was able to resolve this by updating docker-ce and docker-ce-cli to 20.10.8.

**Comment by clp01** [comment]
For me, this was resolved by updating all packages on the system before proceeding with the upgrade.

**Comment by tallsheep** [comment]
well, i found the same problem, and i got fixed it by upgrade docker version form 18.06.1 to 20.10.8.

**Comment by stevenkitter** [comment]
the /data/databse assigned to uid 999 & gid 999.
use `chown 999:999 /data/database`, this can help me. you guys can try.

**Comment by xmskf** [comment]
> well, i found the same problem, and i got fixed it by upgrade docker version form 18.06.1 to 20.10.8.

thank you，This helped me solve the problem~

@wy65701436     docker version is also an important factor

**Comment by CN-Eden** [comment]
If you create an EFS persistent volume, you should change the UID and GID of EFS to be consistent with the UID and GID of PG. For example, change the UID and GID of EFS to 999
